src/prefect/utilities/logging.py

```diff
@@ -92,7 +98,10 @@ def put(self, log: dict) -> None:
             json.dumps(log)  # make sure the payload is serializable
             self.queue.put(log)
         except TypeError as exc:
-            self.logger.critical("Failed to write log with error: {}".format(str(exc)))
+            message = "Failed to write log with error: {}".format(str(exc))
+            self.logger.critical(message)
+
+            self.queue.put(self._make_error_log(message))
```

Comment:
Could this result in an infinite loop if, heaven forbid, Cloud was completely down for an extended period of time?

Reply:
Cloud errors like 400s are surfaced in the above handle. Where an extra check happens  and logs `Unable to write logs to Prefect Cloud`

Reply:
Let me test this again but I believe that's what should happen